Dependencies/xr/Source/ARKit/XR.mm

```diff
@@ -649,14 +651,18 @@ bool TryInitialize() {
                 currentFrame = session.currentFrame;
                 [currentFrame retain];
             }
-            
+
             dispatch_sync(dispatch_get_main_queue(), ^{
                 // Check whether the main view has changed, and if so, reparent the xr view.
                 UIView* currentSuperview = [xrView superview];
                 UIView* desiredSuperview = getMainView();
                 if (currentSuperview != desiredSuperview) {
+                    [currentSuperview release];
                     [xrView removeFromSuperview];
```

Comment:
Does this ordering need to be flipped?

Reply:
Yea that’d be safer.

Reply:
Actually I don't think it matters. The issue I was facing was that the `superview` unexpectedly became `nil`. I think views only keep a weak reference to their superviews, and when a view is deleted, it automatically removes child views. Since calling `removeFromSuperview` is a no-op if there is no superview, then this shouldn't cause any problem.